src/pages/LanguagePreferences.js

```diff
@@ -0,0 +1,152 @@
+import Heading from "./info_page_shared/Heading";
+import Main from "./info_page_shared/Main";
+import Form from "./info_page_shared/Form";
+import FullWidthErrorMsg from "./info_page_shared/FullWidthErrorMsg";
+import FormSection from "./info_page_shared/FormSection";
+import SelectOptions from "./info_page_shared/SelectOptions";
+import ButtonContainer from "./info_page_shared/ButtonContainer";
+import Button from "./info_page_shared/Button";
+import ArrowForwardRoundedIcon from "@mui/icons-material/ArrowForwardRounded";
+
+import validator from "../assorted/validator";
+import LoadingAnimation from "../components/LoadingAnimation";
+
+import { CEFR_LEVELS } from "../assorted/cefrLevels";
+
+export default function LanguagePreferences({ api, setUser }) {
+  const user = useContext(UserContext);
+  const [learned_language, handleLearned_language_change] = useFormField("");
+  const [native_language, handleNative_language_change] = useFormField("en");
+  const [learned_cefr_level, handleLearned_cefr_level_change] =
+    useFormField("");
+  const [systemLanguages, setSystemLanguages] = useState();
+  const [errorMessage, setErrorMessage] = useState("");
+
+  useEffect(() => {
+    api.getSystemLanguages((languages) => {
+      languages.learnable_languages.sort((a, b) => (a.name > b.name ? 1 : -1));
+      languages.native_languages.sort((a, b) => (a.name > b.name ? 1 : -1));
+      setSystemLanguages(languages);
+    });
+    // eslint-disable-next-line
+  }, []);
+
+  if (!systemLanguages) {
+    return <LoadingAnimation />;
+  }
+
+  let validatorRules = [
+    [learned_language === "", "Please select language you want to practice"],
+    [
+      learned_cefr_level === "",
+      "Please select your current level in language you want to practice",
+    ],
+    [
+      native_language === "",
+      "Please select language you want to translations in",
+    ],
+  ];
+
+  let userInfo = {
+    ...user,
+    learned_language: learned_language,
+    learned_cefr_level: learned_cefr_level,
+    native_language: native_language,
+  };
+
+  const modifyCEFRlevel = (languageID, cefrLevel) => {
+    api.modifyCEFRlevel(
+      languageID,
+      cefrLevel,
+      (res) => {
+        console.log("Update '" + languageID + "' CEFR level to: " + cefrLevel);
+        console.log("API returns update status: " + res);
+      },
+      () => {
+        console.log("Connection to server failed...");
+      },
+    );
+  };
+
+  function updateUser(e) {
+    e.preventDefault();
+
+    if (!validator(validatorRules, setErrorMessage)) {
+      return;
+    }
+
+    api.saveUserDetails(userInfo, setErrorMessage, () => {
+      modifyCEFRlevel(learned_language, learned_cefr_level);
+      setUser(userInfo);
+      saveUserInfoIntoCookies(userInfo);
+      redirect("/select_interests");
+    });
+  }
+
+  return (
+    <InfoPage type={"narrow"}>
+      <Header>
+        <Heading>
+          What language would&nbsp;you&nbsp;like&nbsp;to&nbsp;learn?
```

Comment:
Hmm, I remember that @mircealungu mentioned that I should not create string constants because we want to move away from them.  Please correct me if I am wrong.

The non-breaking spaces placed between some words in some headings are to allow more control over typography for various screen sizes and to keep it visually balanced (e.g., a shape of text caused by sentence breaking, readability, and/or preventing orphans (the last word going to the second line of the paragraph) -> here, more about [orphans](https://en.wikipedia.org/wiki/Widows_and_orphans) ). 

Let’s discuss string constants and `&nbsp` during the meeting :)